CHANGELOG.md

```diff
@@ -6,6 +6,43 @@ _**Note:** This is in reverse chronological order, so newer entries are added to
 Swift 5.5
 ---------
 
+* [SR-14731][]:
+
+  The compiler now correctly rejects the application of generic arguments to the
+  special `Self` type:
+
+  ```swift
+  struct Box<T> {
+    // previously interpreted as a return type of Box<T>, ignoring the <Int> part;
+    // now we diagnose an error with a fixit suggesting replacing `Self` with `Box`
+    static func makeBox() -> Self<Int> {}
```

Comment:
This code example should probably have a `Box()` in the body of the function, it seems a bit odd otherwise.